src/main/java/net/wasdev/wlp/ant/types/EmbeddedServerInfo.java

```diff
@@ -0,0 +1,87 @@
+package net.wasdev.wlp.ant.types;
+
+import java.io.File;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class EmbeddedServerInfo {
+    
+    public static class EmbeddedServers {
+        
+        private static final Map<EmbeddedServerInfo, Object> EMBEDDED_SERVERS = new HashMap<EmbeddedServerInfo, Object>();
+        
+        public static Object get(EmbeddedServerInfo info) {
+            // Look for the server based on the server info
+            for (Entry<EmbeddedServerInfo, Object> entry : EMBEDDED_SERVERS.entrySet()) {
+                EmbeddedServerInfo serverInfo = entry.getKey();
+                if (serverInfo.equals(info)) {
+                    return entry.getValue();
+                }
```

Comment:
I may be missing something, but can't you just say `if EMBEDDED_SERVERS.containsKey(info) { return EMBEDDED_SERVERS.get(info) } else { return null; }` instead of looping through the whole entrySet ?

Reply:
I tried something like that but it didn't work. I think it's because it's trying to compare two object references rather than the actual properties of the object. That's why I implemented an `equals` method on `EmbeddedServerInfo` and I invoke it explicitly to determine if they are equal (all the properties must be equal for the two objects to be equal).

Reply:
ok...I saw your override of the equals method and thought the Map would use that...but evidently not from what you are saying.

Reply:
I think the issue is that it's not actually an override...not every `Object` has an `equals` method.